### PR TITLE
fix bug where request body is set to null and native fetch will throw…

### PR DESCRIPTION
**Pull request by u9520107**
… TypeError

### CODE REVIEW

**Comment by kirill-konshin** [comment]
Please fix all tests.

**Comment by u9520107** [comment]
#35 
Fixed the test cases. I mistakenly thought the request object has a 'body' property as well.

Also changed !!init.body back to !init.body.

File: src/http/Client.js [hunk 1]
```diff
@@ -127,7 +128,13 @@ export default class Client extends EventEmitter {
 
             // Assign a new encoded body
             if (contentType.indexOf(ApiResponse._jsonContentType) > -1) {
-                init.body = JSON.stringify(init.body);
+                if((init.method === 'GET' || init.method === 'HEAD') && !!init.body) {
+                    // oddly setting body to null still result in TypeError in phantomjs
+                    init.body = void 0;
```
Comment (kirill-konshin) [inline src/http/Client.js]: What if you just do

```js
delete init.body;
```

Or set it to `undefined`?

Comment (u9520107) [inline src/http/Client.js]: hmm... just set it to undefined I guess. Right, we shouldn't have to worry about undefined being overridden anymore. I'll push another change then.

**Comment by u9520107** [comment]
Is this PR ok?

**Pull request merged**